### PR TITLE
build: Add codecov upload script to repo

### PR DESCRIPTION
**Pull request by imsodin**
Coverage CI jobs are failing since a while (curl missing - huh?). They still show up as successful, because the failure is within a bash command substitution, and those errors aren't passed on. I think the `bash <(curl something)` is ugly anyway (and has been shown as unsafe recently), so I'd prefer having that script in the repo and manually upgrading it (basically whenever uploading breaks, hopefully almost never).

### CODE REVIEW

File: script/codecov-upload.sh [hunk 1]
```diff
@@ -0,0 +1,1883 @@
...
+      if (length($3) > 0) {
+         vn=""; if (indent > 0) {vn=(vn)(vname[0])("_")}
+         printf("%s%s%s=\"%s\"\n", "'"$prefix"'",vn, $2, $3);
+      }
+   }'
+}
+
+if [ $# != 0 ];
+then
+  while getopts "a:A:b:B:cC:dD:e:f:F:g:G:hi:J:k:Kn:p:P:Q:q:r:R:s:S:t:T:u:U:vx:X:Zz:N:-" o
```
Comment (calmh) [inline script/codecov-upload.sh:263]: This could use some comments

File: script/codecov-upload.sh [hunk 2]
```diff
@@ -0,0 +1,1883 @@
...
+else
+  urlencoded_slug=$(urlencode "$slug_o")
+fi
+
+query="branch=$branch\
+       &commit=$commit\
+       &build=$([ "$build_o" = "" ] && echo "$build" || echo "$build_o")\
+       &build_url=$build_url\
+       &name=$(urlencode "$name")\
+       &tag=$([ "$tag_o" = "" ] && echo "$tag" || echo "$tag_o")\
```
Comment (calmh) [inline script/codecov-upload.sh:1116]: Lacks urlencoding

**Review by calmh — CHANGES_REQUESTED** [review]
A bit complex, lacks some tests

**Review by calmh — APPROVED** [review]
Just kidding.

**Comment by imsodin** [comment]
I was wondering why you are reviewing so much on holidays, now I am suspecting you might be are experiencing bit of Audrius' holiday boredom :D

**Pull request merged**